IFC_Engine/Modify/SetIfcProperty.cs

```diff
@@ -0,0 +1,117 @@
+        public static IBHoMObject SetIfcProperty(this IBHoMObject bHoMObject, string propName, object value)
+        {
+            if (bHoMObject == null)
+                return null;
+
+            List<IfcProperty> properties = new List<IfcProperty> { new IfcProperty { Name = propName, Value = value } };
+
+            IfcPropertiesToPush existingFragment = bHoMObject.Fragments.FirstOrDefault(x => x is IfcPropertiesToPush) as IfcPropertiesToPush;
+            if (existingFragment != null)
+            {
+                foreach (IfcProperty prop in existingFragment.Properties)
+                {
+                    if (prop.Name != propName)
+                        properties.Add(prop);
+                }
+            }
+
+            IfcPropertiesToPush fragment = new IfcPropertiesToPush { Properties = properties };
+
+            IBHoMObject obj = bHoMObject.ShallowClone();
+            obj.Fragments = new FragmentSet(bHoMObject.Fragments.Where(x => !(x is IfcPropertiesToPush)).ToList());
+            obj.Fragments.Add(fragment);
+
+            Compute.UnsupportedUnitsWarning();
+
+            return obj;
+        }
+
+        /***************************************************/
+
+        [Description("Attaches properties to a BHoM object, which will be applied to a correspondent IFC element on Push.")]
+        [Input("bHoMObject", "BHoMObject to which the properties will be attached.")]
+        [Input("names", "Names of the properties to be attached.")]
+        [Input("values", "Values of the properties to be attached.")]
+        [Output("bHoMObject")]
+        public static IBHoMObject SetIfcProperties(this IBHoMObject bHoMObject, List<string> propNames, List<object> values)
+        {
+            if (bHoMObject == null)
+                return null;
+
+            if (propNames.Count != values.Count)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Number of input names needs to be equal to the number of input values. Properties have not been set.");
+                return bHoMObject;
+            }
+
+            List<IfcProperty> properties = propNames.Zip(values, (x, y) => new IfcProperty { Name = x, Value = y }).ToList();
+
+            IfcPropertiesToPush existingFragment = bHoMObject.Fragments.FirstOrDefault(x => x is IfcPropertiesToPush) as IfcPropertiesToPush;
+            if (existingFragment != null)
+            {
+                foreach (IfcProperty property in existingFragment.Properties)
+                {
+                    if (propNames.All(x => property.Name != x))
+                        properties.Add(property);
+                }
+            }
+
+            IfcPropertiesToPush fragment = new IfcPropertiesToPush { Properties = properties };
+
+            IBHoMObject obj = bHoMObject.ShallowClone();
+            obj.Fragments = new FragmentSet(bHoMObject.Fragments.Where(x => !(x is IfcPropertiesToPush)).ToList());
+            obj.Fragments.Add(fragment);
```

Comment:
This can be simplified: 
```suggestion
            bHoMObject.Fragments.AddOrReplace(fragment);
```